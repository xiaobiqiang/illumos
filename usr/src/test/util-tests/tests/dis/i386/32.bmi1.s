/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test bmi1 related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	andn	%eax, %ebx, %edx
	andn	(%eax), %ebx, %edx
	andn	0x40(%eax), %ebx, %edx
	bextr	%ebx, %eax, %edx
	bextr	%ebx, (%eax), %edx
	bextr	%ebx, 0x40(%eax), %edx
	blsi	%eax, %edx
	blsi	(%eax), %edx
	blsi	0x40(%eax), %edx
	blsmsk	%eax, %edx
	blsmsk	(%eax), %edx
	blsmsk	0x40(%eax), %edx
	blsr	%eax, %edx
	blsr	(%eax), %edx
	blsr	0x40(%eax), %edx
	tzcnt	%ax, %dx
	tzcnt	(%eax), %dx
	tzcnt	0x40(%eax), %dx
	tzcnt	%eax, %edx
	tzcnt	(%eax), %edx
	tzcnt	0x40(%eax), %edx
.size libdis_test, [.-libdis_test]
