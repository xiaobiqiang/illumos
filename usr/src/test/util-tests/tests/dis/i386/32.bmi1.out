    libdis_test:      c4 e2 60 f2 d0     andn   %eax,%ebx,%edx
    libdis_test+0x5:  c4 e2 60 f2 10     andn   (%eax),%ebx,%edx
    libdis_test+0xa:  c4 e2 60 f2 50 40  andn   0x40(%eax),%ebx,%edx
    libdis_test+0x10: c4 e2 60 f7 d0     bextr  %ebx,%eax,%edx
    libdis_test+0x15: c4 e2 60 f7 10     bextr  %ebx,(%eax),%edx
    libdis_test+0x1a: c4 e2 60 f7 50 40  bextr  %ebx,0x40(%eax),%edx
    libdis_test+0x20: c4 e2 68 f3 d8     blsi   %eax,%edx
    libdis_test+0x25: c4 e2 68 f3 18     blsi   (%eax),%edx
    libdis_test+0x2a: c4 e2 68 f3 58 40  blsi   0x40(%eax),%edx
    libdis_test+0x30: c4 e2 68 f3 d0     blsmsk %eax,%edx
    libdis_test+0x35: c4 e2 68 f3 10     blsmsk (%eax),%edx
    libdis_test+0x3a: c4 e2 68 f3 50 40  blsmsk 0x40(%eax),%edx
    libdis_test+0x40: c4 e2 68 f3 c8     blsr   %eax,%edx
    libdis_test+0x45: c4 e2 68 f3 08     blsr   (%eax),%edx
    libdis_test+0x4a: c4 e2 68 f3 48 40  blsr   0x40(%eax),%edx
    libdis_test+0x50: 66 f3 0f bc d0     tzcnt  %ax,%dx
    libdis_test+0x55: 66 f3 0f bc 10     tzcnt  (%eax),%dx
    libdis_test+0x5a: 66 f3 0f bc 50 40  tzcnt  0x40(%eax),%dx
    libdis_test+0x60: f3 0f bc d0        tzcnt  %eax,%edx
    libdis_test+0x64: f3 0f bc 10        tzcnt  (%eax),%edx
    libdis_test+0x68: f3 0f bc 50 40     tzcnt  0x40(%eax),%edx
